doc/source/io.rst

```diff
@@ -176,6 +176,12 @@ low_memory : boolean, default ``True``
   Note that the entire file is read into a single DataFrame regardless,
   use the ``chunksize`` or ``iterator`` parameter to return the data in chunks.
   (Only valid with C parser)
+buffer_lines : int, default None
+    DEPRECATED: this argument will be removed in a future version because its
+    value is not respected by the parser
+
+    If ``low_memory`` is ``True``, specify the number of rows to be read for
+    each chunk. (Only valid with C parser)
```

Comment:
Can't we leave out the actual explanation? As this is not only deprecated, but also does not work (IIUC), so it does not serve much purpose IMO (apart from explaining what feature exactly has never worked, and will never work ..)

BTW, @gfyoung, for the rest strong +1 on your cleaning up of the keywords!